spydra/src/main/java/com/spotify/spydra/submitter/api/DynamicSubmitter.java

```diff
@@ -213,9 +212,13 @@ private void waitForHistoryToBeMoved(SpydraArgument arguments) throws IOExceptio
     String path = arguments.clusterProperties()
         .getProperty("mapred:mapreduce.jobhistory.intermediate-done-dir");
     URI uri = URI.create(path);
-    FileSystem fs = gcpUtils.fileSystemForUri(uri);
+    String bucketName = uri.getHost();
+    String directory = uri.getPath();
+    if (directory != null && directory.length() > 0) {
+      directory = directory.substring(1, directory.length()); //remove trailing slash from path
+    }
     LOGGER.info("Waiting for history files to be moved to its final location");
-    while (fs.getContentSummary(new Path(uri)).getFileCount() != 0) {
+    while (new StorageUtils().getFileCount(bucketName, directory+"/") != 0) {
```

Comment:
This constructs a new StorageUtils on every iteration.

Reply:
It also pages through the entire object listing on GCS, I don't think there's a concern here though.
Maybe if it's a pooling cluster (thus lots of objects), and the cluster breaks we might spend a lot of time iterating here.

Reply:
Didn't find a method to quickly check the size of the directory. Maybe I should just implement a methos isEmpty() and stop is hasNext() returns anything?

Reply:
I think the abstraction of `StorageUtil` here creates more confusion and conflict between organising code and implementing the logic we would like.